jiant/tasks/seq2seq.py

```diff
@@ -114,24 +118,52 @@ def _make_instance(input_, target):
     def get_metrics(self, reset=False):
         """Get metrics specific to the task"""
         avg_nll = self.scorer1.get_metric(reset)
-        acc = self.scorer2.get_metric(reset)
-        return {"perplexity": math.exp(avg_nll), "accuracy": acc}
+        val_metric = self.scorer2.get_metric(reset)
+        if self.val_metric == "%s_accuracy" % self.name:
+            metric_name = "accuracy"
+        else:  # for MT
+            metric_name = "bleu"
+            val_metric = val_metric["BLEU"]
+        return {"perplexity": math.exp(avg_nll), metric_name: val_metric}
 
     def update_metrics(self, logits, labels, tagmask=None, predictions=None):
         # This doesn't require logits for now, since loss is updated in another part.
         assert logits is None and predictions is not None
 
-        if labels.shape[1] < predictions.shape[2]:
-            predictions = predictions[:, 0, : labels.shape[1]]
+        if self.val_metric != "%s_accuracy" % self.name:  # for MT
+            self.scorer2(predictions[:, 0, :], labels)
         else:
-            predictions = predictions[:, 0, :]
-            # Cut labels if predictions (without gold target) are shorter.
-            labels = labels[:, : predictions.shape[1]]
-            tagmask = tagmask[:, : predictions.shape[1]]
-        self.scorer2(predictions, labels, tagmask)
+            if labels.shape[1] < predictions.shape[2]:
+                predictions = predictions[:, 0, : labels.shape[1]]
+            else:
+                predictions = predictions[:, 0, :]
+                # Cut labels if predictions (without gold target) are shorter.
+                labels = labels[:, : predictions.shape[1]]
+                tagmask = tagmask[:, : predictions.shape[1]]
+            self.scorer2(predictions, labels, tagmask)
         return
 
     def get_prediction(self, voc_src, voc_trg, inputs, gold, output):
+        if self.val_metric == "%s_accuracy" % self.name:
```

Comment:
See above.

Reply:
This part of the code for MT needs to receive the EOS index (related to the above comment regarding BLEU).

Reply:
@pruksmhc

Reply:
Same comment